src/code/z_demo.c

```diff
@@ -1232,683 +1401,778 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Camera_ResetAnim(Play_GetCamera(play, csCtx->subCamId));
-                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->subCamLookAtPoints,
-                                   csCtx->subCamEyePoints, GET_PLAYER(play), relativeToLink);
+                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camLookAtPoints, csCtx->camEyePoints,
+                                   GET_PLAYER(play), relativeToPlayer);
             }
         }
     }
 
     while (shouldContinue) {
-        if (((CutsceneCameraPoint*)cmd)->continueFlag == CS_CMD_STOP) {
-            shouldContinue = 0;
+        if (((CutsceneCameraPoint*)scriptPtr)->continueFlag == CS_CMD_STOP) {
+            shouldContinue = false;
         }
-        cmd += 0x10;
-        size += 0x10;
+
+        scriptPtr += sizeof(CutsceneCameraPoint);
+        size += sizeof(CutsceneCameraPoint);
     }
 
     return size;
 }
 
-// Command 0x7: ? (Related to camera positons)
-s32 Cutscene_Command_07(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 unused) {
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_SetCamEye(PlayState* play, CutsceneContext* csCtx, u8* scriptPtr, u8 unused) {
+    CsCmdGeneric* cmd = (CsCmdGeneric*)scriptPtr;
     s32 size;
-    Vec3f sp3C;
-    Vec3f sp30;
+    Vec3f at;
+    Vec3f eye;
     Camera* subCam;
     f32 sp28;
 
-    cmd += 8;
-    size = 8;
+    scriptPtr += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((D_8015FCC2 < cmdBase->startFrame) || (D_8015FCC2 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            D_8015FCC2 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((gCamEyeAppliedFrame < cmd->startFrame) || (gCamEyeAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)scriptPtr;
+
+        if (csCtx->camAtReady) {
+            // the frame number set here isn't important, it just signals that the camera data has been applied
+            gCamEyeAppliedFrame = cmd->startFrame;
+
+            if (gUseCutsceneCam) {
                 subCam = Play_GetCamera(play, csCtx->subCamId);
                 subCam->player = NULL;
+
                 Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_FREE0);
-                sp28 = csCtx->subCamLookAtPoints->cameraRoll * 1.40625f;
+
+                sp28 = csCtx->camLookAtPoints->cameraRoll * 1.40625f;
                 Camera_SetParam(subCam, 64, &sp28);
```

Comment:
```suggestion
                roll = csCtx->camLookAtPoints->cameraRoll * 1.40625f;
                Camera_SetParam(subCam, 64, &roll);
```
You can name this temp, and that bottom function will look like:
`Camera_SetViewParam(subCam, CAM_VIEW_ROLL, &roll);`
after my cam-flags PR